src/Medology/Behat/CsvContext.php

```diff
@@ -118,4 +119,28 @@ private function assertNoAdditionalHeaders(array $required, array $actual, strin
             throw new Exception("CSV '$csvName' contains extra headers '" . implode("', '", $diff) . "'");
         }
     }
+
+    /**
+     * Update expected data injecting model values.
+     *
+     * @param TableNode $table a table node containing the values to inject
+     *
+     * @throws ReflectionException When injected parameters are invalid
+     */
+    private function updateExpectedDataAndInjectValues(TableNode $table): array
+    {
+        $expectedData = array_map(function ($row) {
+            return array_map(function ($value) {
+                return $this->storeContext->injectStoredValues($value);
+            }, $row);
+        }, $table->getRows());
+
+        $expectedData = array_map(function ($row) {
+            return array_map(function ($value) {
+                return $this->storeContext->injectStoredValues($value);
+            }, $row);
+        }, $table->getRows());
+
+        return array_values($expectedData);
```

Comment:
You could avoid the overhead of assigning and garbage-collecting a variable by using array_values inline:

```suggestion
        return array_values(array_map(function ($row) {
            return array_map(function ($value) {
                return $this->storeContext->injectStoredValues($value);
            }, $row);
        }, $table->getRows()));
```

If you are mapping an array using a function on an object, you can further simplify this and avoid the additional nested function by using an [object method callback](https://www.php.net/manual/en/language.types.callable.php):

```php
        return array_values(array_map(function ($row) {
            return array_map([$this->storeContext, 'injectStoredValues'], $row);
        }, $table->getRows()));
```

Unfortunately, the project currently supports < PHP 7.4, so we can't use [arrow functions](https://www.php.net/manual/en/functions.arrow.php) to simplify it further 😞